frontend/src/app/App.test.tsx

```diff
@@ -230,311 +239,382 @@ describe("App", () => {
-    const dialog = StreamlitDialog({
-      type: DialogType.ABOUT,
-      sessionInfo: mockSessionInfo(),
-      onClose: () => {},
+      const rerunScriptFunc = jest.spyOn(
+        // @ts-expect-error
+        instance.hostCommunicationMgr.props,
+        "rerunScript"
+      )
+
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "RERUN_SCRIPT",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+
+      expect(rerunScriptFunc).toHaveBeenCalledWith()
+      expect(wrapper.state("scriptRunState")).toBe(
+        ScriptRunState.RERUN_REQUESTED
+      )
     })
-    wrapper.setState({ dialog })
-    expect(wrapper.find(Modal)).toHaveLength(1)
-  })
 
-  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when the app is first rendered", () => {
-    const props = getProps()
-    shallow(<App {...props} />)
+    it("fires clearCache function when cacheClearRequested message has been received", () => {
+      instance.isServerConnected = jest.fn().mockReturnValue(true)
 
-    expect(props.hostCommunication.sendMessage).toHaveBeenCalledWith({
-      type: "SCRIPT_RUN_STATE_CHANGED",
-      scriptRunState: ScriptRunState.NOT_RUNNING,
+      const clearCacheFunc = jest.spyOn(
+        // @ts-expect-error
+        instance.hostCommunicationMgr.props,
+        "clearCache"
+      )
+      const closeDialogFunc = jest.spyOn(instance, "closeDialog")
+
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "CLEAR_CACHE",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+
+      expect(clearCacheFunc).toHaveBeenCalledWith()
+      expect(closeDialogFunc).toHaveBeenCalledWith()
     })
-  })
 
-  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when scriptRunState changing", () => {
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
+    it("does not prevent a modal from opening when closure message is set", () => {
+      const dialog = StreamlitDialog({
+        type: DialogType.ABOUT,
+        sessionInfo: mockSessionInfo(),
+        onClose: () => {},
+      })
 
-    Object.values(ScriptRunState).forEach(scriptRunState => {
+      // Open a dialog
+      wrapper.setState({ dialog })
+      expect(wrapper.find(Modal)).toHaveLength(1)
+
+      // Send close message
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "CLOSE_MODAL",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+      expect(wrapper.find(Modal)).toHaveLength(0)
+
+      // Open a dialog again - make sure it works
+      wrapper.setState({ dialog })
+      expect(wrapper.find(Modal)).toHaveLength(1)
+    })
```

Comment:
Hm, it doesn't hurt to have this test, but I think that it's probably unnecessary now that the implementation for how we close an open dialog is less silly. I'm assuming this only existed previously because of how awkward the previous way this was implemented was (it was easy to mess up flipping a boolean flag at some stage)